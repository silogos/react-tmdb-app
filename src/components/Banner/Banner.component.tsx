import { BannerProps } from "./Banner.type";
import { ViewfinderCircleIcon } from "@heroicons/react/24/outline";

function Banner({ src, className }: BannerProps) {
  if (!src)
    return (
      <div
        className={`relative aspect-video bg-gray-700 bg-opacity-90 rounded-lg overflow-hidden ${className}`}
      >
        <div className="absolute inset-0 flex flex-col justify-center items-center gap-2 text-gray-300">
          <ViewfinderCircleIcon className="size-16" />
          Image Not Found
        </div>
      </div>
    );

  return (
    <img
      className={`aspect-video bg-gray-700 bg-opacity-90 rounded-lg ${className}`}
      src={`https://image.tmdb.org/t/p/w500/${src}`}
      alt="Movie Banner"
    ></img>
  );
}

export default Banner;
