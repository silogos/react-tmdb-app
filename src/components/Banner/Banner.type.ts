export type BannerProps = {
  src?: string;
  className?: string;
};
