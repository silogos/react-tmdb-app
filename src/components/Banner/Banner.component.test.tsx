import { describe, it, expect } from "vitest";
import Banner from "./Banner.component";
import { render } from "@testing-library/react";

describe("Banner Component", () => {
  it("renders correctly", () => {
    const component = render(<Banner src="testPath" />);
    expect(component).toBeTruthy();
  });

  it("displays the correct image source", () => {
    const { getByAltText } = render(<Banner src="testPath" />);
    const imgElement = getByAltText("Movie Banner") as HTMLImageElement;
    expect(imgElement.src).toContain(
      "https://image.tmdb.org/t/p/w500/testPath"
    );
  });

  it("displays Image Not Found when src is null", () => {
    const { getByText } = render(<Banner />);
    expect(getByText("Image Not Found")).toBeInTheDocument();
  });
});
