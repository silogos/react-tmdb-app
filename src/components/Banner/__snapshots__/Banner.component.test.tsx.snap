// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Banner Component > displays Image Not Found when src is null 1`] = `
<div>
  <div
    class="relative aspect-video bg-gray-700 bg-opacity-90 rounded-lg overflow-hidden undefined"
  >
    <div
      class="absolute inset-0 flex flex-col justify-center items-center gap-2 text-gray-300"
    >
      <svg
        aria-hidden="true"
        class="size-16"
        data-slot="icon"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.5 3.75H6A2.25 2.25 0 0 0 3.75 6v1.5M16.5 3.75H18A2.25 2.25 0 0 1 20.25 6v1.5m0 9V18A2.25 2.25 0 0 1 18 20.25h-1.5m-9 0H6A2.25 2.25 0 0 1 3.75 18v-1.5M15 12a3 3 0 1 1-6 0 3 3 0 0 1 6 0Z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      Image Not Found
    </div>
  </div>
</div>
`;

exports[`Banner Component > displays the correct image source 1`] = `
<div>
  <img
    alt="Movie Banner"
    class="aspect-video bg-gray-700 bg-opacity-90 rounded-lg undefined"
    src="https://image.tmdb.org/t/p/w500/testPath"
  />
</div>
`;

exports[`Banner Component > renders correctly 1`] = `
<div>
  <img
    alt="Movie Banner"
    class="aspect-video bg-gray-700 bg-opacity-90 rounded-lg undefined"
    src="https://image.tmdb.org/t/p/w500/testPath"
  />
</div>
`;
